identity-android/src/main/java/com/android/identity/SessionEncryption.java

```diff
@@ -51,50 +53,72 @@
 import co.nstant.in.cbor.model.ByteString;
 import co.nstant.in.cbor.model.DataItem;
 import co.nstant.in.cbor.model.Map;
-import co.nstant.in.cbor.model.Number;
 import co.nstant.in.cbor.model.UnicodeString;
 
 
 /**
  * A helper class for encrypting and decrypting messages exchanged with a remote
  * mDL prover, conforming to ISO 18013-5 9.1.1 Session encryption.
  */
-final class SessionEncryptionReader {
+final class SessionEncryption {
 
-    private static final String TAG = "SessionEncryptionReader";
+    private static final String TAG = "SessionEncryption";
+
+    public static final int ROLE_MDOC = 0;
+    public static final int ROLE_MDOC_READER = 1;
+
+    @Retention(RetentionPolicy.SOURCE)
+    @IntDef(value = {ROLE_MDOC, ROLE_MDOC_READER})
+    public @interface Role {
+    }
 
+    protected final @DataTransport.Role int mRole;
     private boolean mSessionEstablishmentSent;
 
-    private final PrivateKey mEReaderKeyPrivate;
+    private final PrivateKey mESenderKeyPrivate;
     private final PublicKey mEReaderKeyPublic;
-    private final PublicKey mEDeviceKeyPublic;
+    private final PublicKey mEReceiverKeyPublic;
 
-    private final SecretKeySpec mSKDevice;
-    private final SecretKeySpec mSKReader;
-    private int mSKDeviceCounter = 1;
-    private int mSKReaderCounter = 1;
+    private final SecretKeySpec mSKReceiver;
+    private final SecretKeySpec mSKSender;
+    private int mSKReceiverCounter = 1;
+    private int mSKSenderCounter = 1;
     private boolean mSendSessionEstablishment = true;
 
     /**
-     * Creates a new {@link SessionEncryptionReader} object.
+     * Creates a new {@link SessionEncryption} object. eReaderKeyPublic is required if you are
+     * making a reader {@link SessionEncryption} object (ie. role is ROLE_MDOC_READER).
```

Comment:
We generally use `{@code xyz}` or `<code>xyz</code>` for things like `eReaderKeyPublic and `ROLE_MDOC_READER` so they stand out in the docs. Same elsewhere.